### PR TITLE
Client synchronize profile settings from sever

### PR DESCRIPTION
**Pull request by lingtaolf**
Fixes:#56976

### Changelog category (leave one):
- Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Clickhouse client will sychronize profile settings from sever while connecting to server.
  

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

### CODE REVIEW

**Comment by robot-clickhouse-ci-1** [comment]
*This is an automated comment for commit 66ad7fec98f6beab4ab009e3aa709a7874bc7256 with description of existing statuses. It's updated for the latest CI running*

[⏳ Click here](https://s3.amazonaws.com/clickhouse-test-reports/58131/66ad7fec98f6beab4ab009e3aa709a7874bc7256/ci_running.html) to open a full report in a separate page

<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/58131/66ad7fec98f6beab4ab009e3aa709a7874bc7256/ci_running.html">⏳ pending</a></td></tr>
<tbody>
</table>

**Comment by lingtaolf** [comment]
@alexey-milovidov Could you give some advices on how to test thoes code ?

**Comment by alexey-milovidov** [comment]
@lingtaolf with an `.sh` stateless test.

CREATE USER ... SETTINGS date_time_input_format = 'best_effort'

Then connect with clickhouse-client using this user and run an INSERT query that should succeed only if `date_time_input_format = 'best_effort'`

See other tests that are creating users.

**Comment by bharatnc** [comment]
Should this behavior be made available through a setting or should this be the default behavior?

**Comment by alexey-milovidov** [comment]
There is a minimal risk of incompatibility - ok to make it the default unconditionally, but if you can easily add a setting, let's do it just in case.

**Comment by lingtaolf** [comment]
@alexey-milovidov Hello, There're some tests failed , Do I need to fix them?

**Comment by alexey-milovidov** [comment]
Yes, they have to be fixed before merging.

**Comment by alexey-milovidov** [comment]
```
==================
WARNING: ThreadSanitizer: data race (pid=755)
  Write of size 1 at 0x7b98000058f8 by thread T755:
    #0 DB::Context::setDistributed(bool) build_docker/./src/Interpreters/Context.h:759:64 (clickhouse+0x1854f063) (BuildId: 4c1cc6f76729ee309d8b0ab4aec6c3f6101bbdd3)
    #1 DB::InterpreterSelectQuery::InterpreterSelectQuery(std::__1::shared_ptr<DB::IAST> const&, std::__1::shared_ptr<DB::Context> const&, std::__1::optional<DB::Pipe>, std::__1::shared_ptr<DB::IStorage> const&, DB::SelectQueryOptions const&, std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>>> const&, std::__1::shared_ptr<DB::StorageInMemoryMetadata const> const&, std::__1::shared_ptr<DB::PreparedSets>)::$_0::operator()(bool) const build_docker/./src/Interpreters/InterpreterSelectQuery.cpp:626:18 (clickhouse+0x1854f063)
    #2 DB::InterpreterSelectQuery::InterpreterSelectQuery(std::__1::shared_ptr<DB::IAST> const&, std::__1::shared_ptr<DB::Context> const&, std::__1::optional<DB::Pipe>, std::__1::shared_ptr<DB::IStorage> const&, DB::SelectQueryOptions const&, std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>>> const&, std::__1::shared_ptr<DB::StorageInMemoryMetadata const> const&, std::__1::shared_ptr<DB::PreparedSets>) build_docker/./src/Interpreters/InterpreterSelectQuery.cpp:783:5 (clickhouse+0x1854a7ff) (BuildId: 4c1cc6f76729ee309d8b0ab4aec6c3f6101bbdd3)
    #3 DB::InterpreterSelectQuery::InterpreterSelectQuery(std::__1::shared_ptr<DB::IAST> const&, std::__1::shared_ptr<DB::Context> const&, DB::SelectQueryOptions const&, std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>>> const&) build_docker/./src/Interpreters/InterpreterSelectQuery.cpp:209:7 (clickhouse+0x185462c5) (BuildId: 4c1cc6f76729ee309d8b0ab4aec6c3f6101bbdd3)
    #4 std::__1::__unique_if<DB::InterpreterSelectQuery>::__unique_single std::__1::make_unique[abi:v15000]<DB::InterpreterSelectQuery, std::__1::shared_ptr<DB::IAST> const&, std::__1::shared_ptr<DB::Context>&, DB::SelectQueryOptions&, std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>>> const&>(std::__1::shared_ptr<DB::IAST> const&, std::__1::shared_ptr<DB::Context>&, DB::SelectQueryOptions&, std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>>> const&) build_docker/./contrib/llvm-project/libcxx/include/__memory/unique_ptr.h:714:32 (clickhouse+0x1863a073) (BuildId: 4c1cc6f76729ee309d8b0ab4aec6c3f6101bbdd3)
    #5 DB::InterpreterSelectWithUnionQuery::buildCurrentChildInterpreter(std::__1::shared_ptr<DB::IAST> const&, std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>>> const&) build_docker/./src/Interpreters/InterpreterSelectWithUnionQuery.cpp:254:16 (clickhouse+0x1863a073)
    #6 DB::InterpreterSelectWithUnionQuery::InterpreterSelectWithUnionQuery(std::__1::shared_ptr<DB::IAST> const&, std::__1::shared_ptr<DB::Context>, DB::SelectQueryOptions const&, std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>>> const&) build_docker/./src/Interpreters/InterpreterSelectWithUnionQuery.cpp:152:13 (clickhouse+0x186371aa) (BuildId: 4c1cc6f76729ee309d8b0ab4aec6c3f6101bbdd3)
    #7 std::__1::__unique_if<DB::InterpreterSelectWithUnionQuery>::__unique_single std::__1::make_unique[abi:v15000]<DB::InterpreterSelectWithUnionQuery, std::__1::shared_ptr<DB::IAST>&, std::__1::shared_ptr<DB::Context>&, DB::SelectQueryOptions const&>(std::__1::shared_ptr<DB::IAST>&, std::__1::shared_ptr<DB::Context>&, DB::SelectQueryOptions const&) build_docker/./contrib/llvm-project/libcxx/include/__memory/unique_ptr.h:714:32 (clickhouse+0x184e1529) (BuildId: 4c1cc6f76729ee309d8b0ab4aec6c3f6101bbdd3)
    #8 DB::InterpreterFactory::get(std::__1::shared_ptr<DB::IAST>&, std::__1::shared_ptr<DB::Context>, DB::SelectQueryOptions const&) build_docker/./src/Interpreters/InterpreterFactory.cpp:166:16 (clickhouse+0x184e03f6) (BuildId: 4c1cc6f76729ee309d8b0ab4aec6c3f6101bbdd3)
    #9 DB::executeQueryImpl(char const*, char const*, std::__1::shared_ptr<DB::Context>, DB::QueryFlags, DB::QueryProcessingStage::Enum, DB::ReadBuffer*) build_docker/./src/Interpreters/executeQuery.cpp:1046:31 (clickhouse+0x18add69e) (BuildId: 4c1cc6f76729ee309d8b0ab4aec6c3f6101bbdd3)
    #10 DB::executeQuery(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>> const&, std::__1::shared_ptr<DB::Context>, DB::QueryFlags, DB::QueryProcessingStage::Enum) build_docker/./src/Interpreters/executeQuery.cpp:1286:26 (clickhouse+0x18ad99b7) (BuildId: 4c1cc6f76729ee309d8b0ab4aec6c3f6101bbdd3)
    #11 DB::TCPHandler::sendHello() build_docker/./src/Server/TCPHandler.cpp:1572:34 (clickhouse+0x1a0a5e70) (BuildId: 4c1cc6f76729ee309d8b0ab4aec6c3f6101bbdd3)
    #12 DB::TCPHandler::runImpl() build_docker/./src/Server/TCPHandler.cpp:266:9 (clickhouse+0x1a09b857) (BuildId: 4c1cc6f76729ee309d8b0ab4aec6c3f6101bbdd3)
    #13 DB::TCPHandler::run() build_docker/./src/Server/TCPHandler.cpp:2364:9 (clickhouse+0x1a0bbe07) (BuildId: 4c1cc6f76729ee309d8b0ab4aec6c3f6101bbdd3)
    #14 Poco::Net::TCPServerConnection::start() build_docker/./base/poco/Net/src/TCPServerConnection.cpp:43:3 (clickhouse+0x1dc4d7e2) (BuildId: 4c1cc6f76729ee309d8b0ab4aec6c3f6101bbdd3)
    #15 Poco::Net::TCPServerDispatcher::run() build_docker/./base/poco/Net/src/TCPServerDispatcher.cpp:115:20 (clickhouse+0x1dc4e05a) (BuildId: 4c1cc6f76729ee309d8b0ab4aec6c3f6101bbdd3)
    #16 Poco::PooledThread::run() build_docker/./base/poco/Foundation/src/ThreadPool.cpp:188:14 (clickhouse+0x1de4de06) (BuildId: 4c1cc6f76729ee309d8b0ab4aec6c3f6101bbdd3)
    #17 Poco::(anonymous namespace)::RunnableHolder::run() build_docker/./base/poco/Foundation/src/Thread.cpp:45:11 (clickhouse+0x1de4c0cf) (BuildId: 4c1cc6f76729ee309d8b0ab4aec6c3f6101bbdd3)
    #18 Poco::ThreadImpl::runnableEntry(void*) build_docker/./base/poco/Foundation/src/Thread_POSIX.cpp:335:27 (clickhouse+0x1de4a589) (BuildId: 4c1cc6f76729ee309d8b0ab4aec6c3f6101bbdd3)

  Previous read of size 1 at 0x7b98000058f8 by thread T771:
    #0 DB::ContextData::ContextData(DB::ContextData const&) build_docker/./src/Interpreters/Context.cpp:758:14 (clickhouse+0x178432f0) (BuildId: 4c1cc6f76729ee309d8b0ab4aec6c3f6101bbdd3)
    #1 DB::Context::Context(DB::Context const&) build_docker/./src/Interpreters/Context.cpp:761:41 (clickhouse+0x17844ed5) (BuildId: 4c1cc6f76729ee309d8b0ab4aec6c3f6101bbdd3)
    #2 DB::Context::createCopy(std::__1::shared_ptr<DB::Context const> const&) build_docker/./src/Interpreters/Context.cpp:795:41 (clickhouse+0x17844ed5)
    #3 DB::(anonymous namespace)::getSubqueryContext(std::__1::shared_ptr<DB::Context const> const&) build_docker/./src/Interpreters/InterpreterSelectQuery.cpp:249:29 (clickhouse+0x1854e8a5) (BuildId: 4c1cc6f76729ee309d8b0ab4aec6c3f6101bbdd3)
    #4 DB::InterpreterSelectQuery::InterpreterSelectQuery(std::__1::shared_ptr<DB::IAST> const&, std::__1::shared_ptr<DB::Context> const&, std::__1::optional<DB::Pipe>, std::__1::shared_ptr<DB::IStorage> const&, DB::SelectQueryOptions const&, std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>>> const&, std::__1::shared_ptr<DB::StorageInMemoryMetadata const> const&, std::__1::shared_ptr<DB::PreparedSets>) build_docker/./src/Interpreters/InterpreterSelectQuery.cpp:433:32 (clickhouse+0x18547015) (BuildId: 4c1cc6f76729ee309d8b0ab4aec6c3f6101bbdd3)
    #5 DB::InterpreterSelectQuery::InterpreterSelectQuery(std::__1::shared_ptr<DB::IAST> const&, std::__1::shared_ptr<DB::Context> const&, DB::SelectQueryOptions const&, std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>>> const&) build_docker/./src/Interpreters/InterpreterSelectQuery.cpp:209:7 (clickhouse+0x185462c5) (BuildId: 4c1cc6f76729ee309d8b0ab4aec6c3f6101bbdd3)
    #6 std::__1::__unique_if<DB::InterpreterSelectQuery>::__unique_single std::__1::make_unique[abi:v15000]<DB::InterpreterSelectQuery, std::__1::shared_ptr<DB::IAST> const&, std::__1::shared_ptr<DB::Context>&, DB::SelectQueryOptions&, std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>>> const&>(std::__1::shared_ptr<DB::IAST> const&, std::__1::shared_ptr<DB::Context>&, DB::SelectQueryOptions&, std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>>> const&) build_docker/./contrib/llvm-project/libcxx/include/__memory/unique_ptr.h:714:32 (clickhouse+0x1863a073) (BuildId: 4c1cc6f76729ee309d8b0ab4aec6c3f6101bbdd3)
    #7 DB::InterpreterSelectWithUnionQuery::buildCurrentChildInterpreter(std::__1::shared_ptr<DB::IAST> const&, std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>>> const&) build_docker/./src/Interpreters/InterpreterSelectWithUnionQuery.cpp:254:16 (clickhouse+0x1863a073)
    #8 DB::InterpreterSelectWithUnionQuery::InterpreterSelectWithUnionQuery(std::__1::shared_ptr<DB::IAST> const&, std::__1::shared_ptr<DB::Context>, DB::SelectQueryOptions const&, std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>>> const&) build_docker/./src/Interpreters/InterpreterSelectWithUnionQuery.cpp:152:13 (clickhouse+0x186371aa) (BuildId: 4c1cc6f76729ee309d8b0ab4aec6c3f6101bbdd3)
    #9 std::__1::__unique_if<DB::InterpreterSelectWithUnionQuery>::__unique_single std::__1::make_unique[abi:v15000]<DB::InterpreterSelectWithUnionQuery, std::__1::shared_ptr<DB::IAST>&, std::__1::shared_ptr<DB::Context>&, DB::SelectQueryOptions const&>(std::__1::shared_ptr<DB::IAST>&, std::__1::shared_ptr<DB::Context>&, DB::SelectQueryOptions const&) build_docker/./contrib/llvm-project/libcxx/include/__memory/unique_ptr.h:714:32 (clickhouse+0x184e1529) (BuildId: 4c1cc6f76729ee309d8b0ab4aec6c3f6101bbdd3)
    #10 DB::InterpreterFactory::get(std::__1::shared_ptr<DB::IAST>&, std::__1::shared_ptr<DB::Context>, DB::SelectQueryOptions const&) build_docker/./src/Interpreters/InterpreterFactory.cpp:166:16 (clickhouse+0x184e03f6) (BuildId: 4c1cc6f76729ee309d8b0ab4aec6c3f6101bbdd3)
    #11 DB::executeQueryImpl(char const*, char const*, std::__1::shared_ptr<DB::Context>, DB::QueryFlags, DB::QueryProcessingStage::Enum, DB::ReadBuffer*) build_docker/./src/Interpreters/executeQuery.cpp:1046:31 (clickhouse+0x18add69e) (BuildId: 4c1cc6f76729ee309d8b0ab4aec6c3f6101bbdd3)
    #12 DB::executeQuery(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>> const&, std::__1::shared_ptr<DB::Context>, DB::QueryFlags, DB::QueryProcessingStage::Enum) build_docker/./src/Interpreters/executeQuery.cpp:1286:26 (clickhouse+0x18ad99b7) (BuildId: 4c1cc6f76729ee309d8b0ab4aec6c3f6101bbdd3)
    #13 DB::TCPHandler::sendHello() build_docker/./src/Server/TCPHandler.cpp:1572:34 (clickhouse+0x1a0a5e70) (BuildId: 4c1cc6f76729ee309d8b0ab4aec6c3f6101bbdd3)
    #14 DB::TCPHandler::runImpl() build_docker/./src/Server/TCPHandler.cpp:266:9 (clickhouse+0x1a09b857) (BuildId: 4c1cc6f76729ee309d8b0ab4aec6c3f6101bbdd3)
    #15 DB::TCPHandler::run() build_docker/./src/Server/TCPHandler.cpp:2364:9 (clickhouse+0x1a0bbe07) (BuildId: 4c1cc6f76729ee309d8b0ab4aec6c3f6101bbdd3)
    #16 Poco::Net::TCPServerConnection::start() build_docker/./base/poco/Net/src/TCPServerConnection.cpp:43:3 (clickhouse+0x1dc4d7e2) (BuildId: 4c1cc6f76729ee309d8b0ab4aec6c3f6101bbdd3)
    #17 Poco::Net::TCPServerDispatcher::run() build_docker/./base/poco/Net/src/TCPServerDispatcher.cpp:115:20 (clickhouse+0x1dc4e05a) (BuildId: 4c1cc6f76729ee309d8b0ab4aec6c3f6101bbdd3)
    #18 Poco::PooledThread::run() build_docker/./base/poco/Foundation/src/ThreadPool.cpp:188:14 (clickhouse+0x1de4de06) (BuildId: 4c1cc6f76729ee309d8b0ab4aec6c3f6101bbdd3)
    #19 Poco::(anonymous namespace)::RunnableHolder::run() build_docker/./base/poco/Foundation/src/Thread.cpp:45:11 (clickhouse+0x1de4c0cf) (BuildId: 4c1cc6f76729ee309d8b0ab4aec6c3f6101bbdd3)
    #20 Poco::ThreadImpl::runnableEntry(void*) build_docker/./base/poco/Foundation/src/Thread_POSIX.cpp:335:27 (clickhouse+0x1de4a589) (BuildId: 4c1cc6f76729ee309d8b0ab4aec6c3f6101bbdd3)

  Location is heap block of size 11904 at 0x7b9800003000 allocated by main thread:
    #0 operator new(unsigned long, std::align_val_t) <null> (clickhouse+0x72fda1b) (BuildId: 4c1cc6f76729ee309d8b0ab4aec6c3f6101bbdd3)
    #1 DB::Context::createGlobal(DB::ContextSharedPart*) build_docker/./src/Interpreters/Context.cpp:774:41 (clickhouse+0x17844be7) (BuildId: 4c1cc6f76729ee309d8b0ab4aec6c3f6101bbdd3)
    #2 DB::Server::main(std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>>> const&) build_docker/./programs/server/Server.cpp:671:22 (clickhouse+0xf63e975) (BuildId: 4c1cc6f76729ee309d8b0ab4aec6c3f6101bbdd3)
    #3 Poco::Util::Application::run() build_docker/./base/poco/Util/src/Application.cpp:315:8 (clickhouse+0x1dc670fe) (BuildId: 4c1cc6f76729ee309d8b0ab4aec6c3f6101bbdd3)
    #4 DB::Server::run() build_docker/./programs/server/Server.cpp:399:25 (clickhouse+0xf63c421) (BuildId: 4c1cc6f76729ee309d8b0ab4aec6c3f6101bbdd3)
    #5 Poco::Util::ServerApplication::run(int, char**) build_docker/./base/poco/Util/src/ServerApplication.cpp:131:9 (clickhouse+0x1dc86e54) (BuildId: 4c1cc6f76729ee309d8b0ab4aec6c3f6101bbdd3)
    #6 mainEntryClickHouseServer(int, char**) build_docker/./programs/server/Server.cpp:205:20 (clickhouse+0xf639583) (BuildId: 4c1cc6f76729ee309d8b0ab4aec6c3f6101bbdd3)
    #7 main build_docker/./programs/main.cpp:505:12 (clickhouse+0x72ff8e0) (BuildId: 4c1cc6f76729ee309d8b0ab4aec6c3f6101bbdd3)

  Thread T755 'TCPHandler' (tid=1594, running) created by thread T740 at:
    #0 pthread_create <null> (clickhouse+0x72774ab) (BuildId: 4c1cc6f76729ee309d8b0ab4aec6c3f6101bbdd3)
    #1 Poco::ThreadImpl::startImpl(Poco::SharedPtr<Poco::Runnable, Poco::ReferenceCounter, Poco::ReleasePolicy<Poco::Runnable>>) build_docker/./base/poco/Foundation/src/Thread_POSIX.cpp:198:6 (clickhouse+0x1de49f3f) (BuildId: 4c1cc6f76729ee309d8b0ab4aec6c3f6101bbdd3)
    #2 Poco::Thread::start(Poco::Runnable&) build_docker/./base/poco/Foundation/src/Thread.cpp:118:2 (clickhouse+0x1de4bacf) (BuildId: 4c1cc6f76729ee309d8b0ab4aec6c3f6101bbdd3)
    #3 Poco::PooledThread::start() build_docker/./base/poco/Foundation/src/ThreadPool.cpp:78:10 (clickhouse+0x1de4fe4b) (BuildId: 4c1cc6f76729ee309d8b0ab4aec6c3f6101bbdd3)
    #4 Poco::ThreadPool::getThread() build_docker/./base/poco/Foundation/src/ThreadPool.cpp:446:14 (clickhouse+0x1de4fe4b)
    #5 Poco::ThreadPool::startWithPriority(Poco::Thread::Priority, Poco::Runnable&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>> const&) build_docker/./base/poco/Foundation/src/ThreadPool.cpp:350:2 (clickhouse+0x1de50247) (BuildId: 4c1cc6f76729ee309d8b0ab4aec6c3f6101bbdd3)
    #6 Poco::Net::TCPServerDispatcher::enqueue(Poco::Net::StreamSocket const&) build_docker/./base/poco/Net/src/TCPServerDispatcher.cpp:152:17 (clickhouse+0x1dc4e56b) (BuildId: 4c1cc6f76729ee309d8b0ab4aec6c3f6101bbdd3)
    #7 Poco::Net::TCPServer::run() build_docker/./base/poco/Net/src/TCPServer.cpp:148:21 (clickhouse+0x1dc4d0d7) (BuildId: 4c1cc6f76729ee309d8b0ab4aec6c3f6101bbdd3)
    #8 Poco::(anonymous namespace)::RunnableHolder::run() build_docker/./base/poco/Foundation/src/Thread.cpp:45:11 (clickhouse+0x1de4c0cf) (BuildId: 4c1cc6f76729ee309d8b0ab4aec6c3f6101bbdd3)
    #9 Poco::ThreadImpl::runnableEntry(void*) build_docker/./base/poco/Foundation/src/Thread_POSIX.cpp:335:27 (clickhouse+0x1de4a589) (BuildId: 4c1cc6f76729ee309d8b0ab4aec6c3f6101bbdd3)

  Thread T771 'TCPHandler' (tid=1638, running) created by thread T740 at:
    #0 pthread_create <null> (clickhouse+0x72774ab) (BuildId: 4c1cc6f76729ee309d8b0ab4aec6c3f6101bbdd3)
    #1 Poco::ThreadImpl::startImpl(Poco::SharedPtr<Poco::Runnable, Poco::ReferenceCounter, Poco::ReleasePolicy<Poco::Runnable>>) build_docker/./base/poco/Foundation/src/Thread_POSIX.cpp:198:6 (clickhouse+0x1de49f3f) (BuildId: 4c1cc6f76729ee309d8b0ab4aec6c3f6101bbdd3)
    #2 Poco::Thread::start(Poco::Runnable&) build_docker/./base/poco/Foundation/src/Thread.cpp:118:2 (clickhouse+0x1de4bacf) (BuildId: 4c1cc6f76729ee309d8b0ab4aec6c3f6101bbdd3)
    #3 Poco::PooledThread::start() build_docker/./base/poco/Foundation/src/ThreadPool.cpp:78:10 (clickhouse+0x1de4fe4b) (BuildId: 4c1cc6f76729ee309d8b0ab4aec6c3f6101bbdd3)
    #4 Poco::ThreadPool::getThread() build_docker/./base/poco/Foundation/src/ThreadPool.cpp:446:14 (clickhouse+0x1de4fe4b)
    #5 Poco::ThreadPool::startWithPriority(Poco::Thread::Priority, Poco::Runnable&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>> const&) build_docker/./base/poco/Foundation/src/ThreadPool.cpp:350:2 (clickhouse+0x1de50247) (BuildId: 4c1cc6f76729ee309d8b0ab4aec6c3f6101bbdd3)
    #6 Poco::Net::TCPServerDispatcher::enqueue(Poco::Net::StreamSocket const&) build_docker/./base/poco/Net/src/TCPServerDispatcher.cpp:152:17 (clickhouse+0x1dc4e56b) (BuildId: 4c1cc6f76729ee309d8b0ab4aec6c3f6101bbdd3)
    #7 Poco::Net::TCPServer::run() build_docker/./base/poco/Net/src/TCPServer.cpp:148:21 (clickhouse+0x1dc4d0d7) (BuildId: 4c1cc6f76729ee309d8b0ab4aec6c3f6101bbdd3)
    #8 Poco::(anonymous namespace)::RunnableHolder::run() build_docker/./base/poco/Foundation/src/Thread.cpp:45:11 (clickhouse+0x1de4c0cf) (BuildId: 4c1cc6f76729ee309d8b0ab4aec6c3f6101bbdd3)
    #9 Poco::ThreadImpl::runnableEntry(void*) build_docker/./base/poco/Foundation/src/Thread_POSIX.cpp:335:27 (clickhouse+0x1de4a589) (BuildId: 4c1cc6f76729ee309d8b0ab4aec6c3f6101bbdd3)

SUMMARY: ThreadSanitizer: data race build_docker/./src/Interpreters/Context.h:759:64 in DB::Context::setDistributed(bool)
==================
```

**Comment by alexey-milovidov** [comment]
@lingtaolf, let's continue.

**Comment by lingtaolf** [comment]
> @lingtaolf, let's continue.

@alexey-milovidov ok

File: src/Client/Connection.cpp [hunk 1]
```diff
@@ -430,6 +434,48 @@ void Connection::receiveHello(const Poco::Timespan & handshake_timeout)
             readIntBinary(read_nonce, *in);
             nonce.emplace(read_nonce);
         }
+
+        //Synchronize settings from server
+        size_t settings_size = 0;
+        if (client_name == "client")
```
[the remark this answers is not in the document]
Comment (lingtaolf) [reply src/Client/Connection.cpp:440]: Currently, It's only used with clickhouse-client.

**Comment by Blargian** [comment]
@lingtaolf you might get some more CI tests passing with a merge of master branch into your branch (as long as there are no conflicts). I know there were a few tests last week which broke on the CI, CH team seems to have done some changes to CI scripts to fix those.

**Comment by lingtaolf** [comment]
> @lingtaolf you might get some more CI tests passing with a merge of master branch into your branch (as long as there are no conflicts). I know there were a few tests last week which broke on the CI, CH team seems to have done some changes to CI scripts to fix those.

Thx!

**Comment by alexey-milovidov** [comment]
@lingtaolf, this feature is very appreciated, let's continue!

**Comment by alexey-milovidov** [comment]
If the client protocol's version is new enough to support server settings, it should send them in the server's Hello packet. The client will receive this packet and apply the server's settings, then apply client-side settings on top of them.

**Comment by alexey-milovidov** [comment]
@lingtaolf, why closing?

**Comment by al13n321** [comment]
https://github.com/ClickHouse/ClickHouse/pull/71178

**Pull request closed without merging**